### ISSUE TITLE
Compatibility with GridSearchCV

### ISSUE THREAD

**Issue by anirudt** [all-umass/metric-learn#34]
Hi,

I wanted to know if there is a possibility in the future for a compatibility for our metric learning algorithms with GridSearchCV in the pipeline. This is wrt the following 

http://scikit-learn.org/dev/developers/contributing.html#rolling-your-own-estimator

**Comment by perimosocordiae** [comment]
That would be nice!

**Comment by svecon** [comment]
This already works since version '0.2.1'.

I would encourage you to use my HierarchicalGridSearchCV for this where you can save a lot of time if you have a long Pipeline. https://github.com/svecon/HierarchicalGridSearchCV

If you still want to use the scikit version for this (warning: inefficient) then example code could look like this:
```
from sklearn.model_selection import GridSearchCV
from sklearn import svm, datasets
from sklearn.pipeline import Pipeline
from metric_learn import LFDA

iris = datasets.load_iris()
clf = GridSearchCV(
    Pipeline([('lfda', LFDA()), ('svr', svm.SVC())]),
    {'lfda__k':(1, 3), 'svr__C':(.1, .5)}
)
clf.fit(iris.data, iris.target)
print(clf.best_score_, clf.best_params_)
```
